src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,796 @@
+                    silent=True, random_state=self.seed
+                )
+            except ImportError:
+                raise ImportError("Please install catboost package to use CatBoostClassifier")
+        elif classifier_name == "XGBClassifier":
+            try:
+                import xgboost
+
+                self.requirements["xgboost"] = xgboost.__version__
+                classifier_mapping["XGBClassifier"] = xgboost.XGBClassifier(
+                    use_label_encoder=False, random_state=self.seed
+                )
+            except ImportError:
+                raise ImportError("Please install xgboost package to use XGBClassifier")
+
+        if classifier_name not in classifier_mapping:
+            raise ValueError(
+                f"Unknown classifier: {classifier_name}. To see list of supported classifiers run\n\t>>> from spikeinterface.curation import default_classifier_search_spaces\n\t>>> print(default_classifier_search_spaces.keys())"
+            )
+
+        return classifier_mapping[classifier_name]
+
+    def get_classifier_search_space(self, classifier_name):
+
+        if classifier_name not in default_classifier_search_spaces:
+            raise ValueError(
+                f"Unknown classifier: {classifier_name}. To see list of supported classifiers run\n\t>>> from spikeinterface.curation import default_classifier_search_spaces\n\t>>> print(default_classifier_search_spaces.keys())"
+            )
+
+        model = self.get_classifier_instance(classifier_name)
+        if self.classifier_search_space is not None:
+            param_space = self.classifier_search_space[classifier_name]
+        else:
+            param_space = default_classifier_search_spaces[classifier_name]
+        return model, param_space
+
+    def evaluate_model_config(self):
+        """
+        Evaluates the model configurations with the given imputation strategies, scaling techniques, and classifiers.
+
+        This method splits the preprocessed data into training and testing sets, then evaluates the specified
+        combinations of imputation strategies, scaling techniques, and classifiers. The evaluation results are
+        saved to the output folder.
+
+        Raises
+        ------
+        ValueError
+            If any of the specified classifier names are not recognized.
+
+        Notes
+        -----
+        The method converts the classifier names to actual classifier instances before evaluating them.
+        The evaluation results, including the best model and its parameters, are saved to the output folder.
+        """
+        from sklearn.model_selection import train_test_split
+
+        X_train, X_test, y_train, y_test = train_test_split(
+            self.X, self.y, test_size=self.test_size, random_state=self.seed, stratify=self.y
+        )
+        classifier_instances = [self.get_classifier_instance(clf) for clf in self.classifiers]
+        self._evaluate(
```

Comment:
There might be. It already does quite a lot: there are lots of classifier instances because it tries out different hyperparameters of the classifiers. Someone who knows more might be able to comment @anoushkajain @alejoe91